### ISSUE TITLE
NewExpression assumes the result of GetGetMethod will be non-null

### ISSUE THREAD

**Issue by tuespetre** [dotnet/corefx#29463]
If the `GetMethod` is non-public, `GetGetMethod()` will not return it, and a `NullReferenceException` can occur.

https://github.com/dotnet/corefx/blob/ff7aca470c71e5bcc00f09147b1c009b3944863c/src/System.Linq.Expressions/src/System/Linq/Expressions/NewExpression.cs#L316

**Comment by svick** [comment]
If I understand it correctly, you're saying that the following code should not throw `NullReferenceException`, right?

```c#
class C
{
    public C(int p) {}

    private int P { get; }
}

void Main()
{
    Expression.New(
        typeof(C).GetConstructors().Single(), new[] { Expression.Constant(42) },
        new[] { typeof(C).GetProperty("P", BindingFlags.Instance | BindingFlags.NonPublic) });
}
```

Though I don't understand your suggested fix. The caller can indeed choose to pass in a `MethodInfo` for the non-public getter, but this issue happens when it doesn't do that. My suggested fix would be to use `GetGetMethod(nonPublic: true)` here.

**Comment by tuespetre** [comment]
@svick that is correct. I was just mentioning the ability to pass in the `MethodInfo` to show that we already have the ability to give the `NewExpression` a `PropertyInfo` with a non-public getter. My suggested fix was using `GetMethod` instead of `GetGetMethod()`, and `GetMethod` calls into `GetGetMethod(nonPublic: true)` so it looks like we were on the same page but a different word 😉
